### PR TITLE
Search component

### PR DESCRIPTION
**Pull request by dwantula**

### CODE REVIEW

File: src/App.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.App {
```
Comment (mi-chal) [inline src/App.scss]: I guess this is not appropriate class name :)

File: src/shared/components/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+
+interface Button {
```
Comment (mi-chal) [inline src/shared/components/Button/Button.tsx]: Create separate file in`Button` folder called `types.ts`. You should keep all the types there.
This should be:
 ```
 export type Props = { 
   className: string;
   onClick: React.MouseEventHandler<HTMLButtonElement>;
   buttonText: string;
 }
 ```

File: src/shared/components/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  className: string;
+  onClick: React.MouseEventHandler<HTMLButtonElement>;
+  buttonText: string;
+}
+
+const Button: React.FC<Button> = ({
```
Comment (mi-chal) [inline src/shared/components/Button/Button.tsx]: So are you using const or function naming convention? Stick with chosen one.

File: src/shared/components/Button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  buttonText,
+}: Button) => {
+  return (
+    <button type="button" className={className} onClick={onClick}>
+      {buttonText}
+    </button>
+  );
+};
+
+export default Button;
```
Comment (mi-chal) [inline src/shared/components/Button/Button.tsx]: From now export all components with memoization:
`export default memo(Button)`
Read about memo - thanks to memoization component will not re-render if props don't change.

File: src/shared/components/SearchInput/SearchInput.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+import Input from '../Input/Input';
+import Button from '../Button/Button';
+
+import './styles.scss';
+
+const places: string[] = ['Anglia', 'Polska', 'Hiszpania', 'Włochy'];
+
+function SearchInput(): JSX.Element {
+  const [inputValue, setInputValue] = useState<string>('');
```
Comment (mi-chal) [inline src/shared/components/SearchInput/SearchInput.tsx]: Read about type inference: https://www.typescriptlang.org/docs/handbook/type-inference.html
You do not need to add `<string>` if you have initialised with `''` - TS will know this is a string

File: src/shared/components/SearchInput/SearchInput.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react';
+import debounce from 'lodash.debounce';
+
+import Input from '../Input/Input';
+import Button from '../Button/Button';
+
+import './styles.scss';
+
+const places: string[] = ['Anglia', 'Polska', 'Hiszpania', 'Włochy'];
```
Comment (mi-chal) [inline src/shared/components/SearchInput/SearchInput.tsx]: Remove

File: src/shared/components/SearchInput/SearchInput.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+function SearchInput(): JSX.Element {
+  const [inputValue, setInputValue] = useState<string>('');
+
+  function searchPlaces() {
+    console.log(inputValue);
+  }
```
Comment (mi-chal) [inline src/shared/components/SearchInput/SearchInput.tsx]: Remove

File: src/shared/components/SearchInput/SearchInput.tsx [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+        placeholder="Whrite city, road"
+        value={inputValue}
+        onChange={handleInputChange}
+      />
+      <Button onClick={searchPlaces} className="button" buttonText="search" />
+      {places.map((elem) => (
+        <ul key={elem}>
+          <li>{elem}</li>
+        </ul>
+      ))}
```
Comment (mi-chal) [inline src/shared/components/SearchInput/SearchInput.tsx]: Remove

**Review by mi-chal — COMMENTED** [review]
Add debounce logic

File: src/shared/components/SearchInput/SearchInput.tsx (hunk 8 above)
Comment (dwantula) [reply src/shared/components/SearchInput/SearchInput.tsx]: Done

File: src/shared/components/SearchInput/SearchInput.tsx (hunk 7 above)
Comment (dwantula) [reply src/shared/components/SearchInput/SearchInput.tsx]: Done

File: src/shared/components/SearchInput/SearchInput.tsx (hunk 6 above)
Comment (dwantula) [reply src/shared/components/SearchInput/SearchInput.tsx]: Done

File: src/shared/components/SearchInput/SearchInput.tsx (hunk 5 above)
Comment (dwantula) [reply src/shared/components/SearchInput/SearchInput.tsx]: ok Fixed

File: src/shared/components/Button/Button.tsx (hunk 4 above)
Comment (dwantula) [reply src/shared/components/Button/Button.tsx]: Fixed

File: src/shared/components/Button/Button.tsx (hunk 3 above)
Comment (dwantula) [reply src/shared/components/Button/Button.tsx]: ok Fixed

File: src/shared/components/Button/Button.tsx (hunk 2 above)
Comment (dwantula) [reply src/shared/components/Button/Button.tsx]: Done

File: src/App.scss (hunk 1 above)
Comment (dwantula) [reply src/App.scss]: No, i wanted try Sass package. I removed it.

**Pull request merged**